### PR TITLE
Bugfix: Don't allow multiple subscriptions for the same runlog job

### PR DESCRIPTION
**Pull request by connorwstein**
Arose from the following scenario:
1. We read too much from the geth websocket
2. Leads to errors in many places, one of which is the head tracker itself. We see "Error in new head subscription, unsubscribed: websocket: read limit exceeded" in the logs
3. We resubscribe with subscribeToHead, which triggers a new Connect call on all head trackables including the jobSubscriber
4. The jobSubscribers Connect will call AddJob again for all the jobs which creates new subscriptions. AddJob does NOT currently check if it already has a subscription for that job, just overwrites it
5. The listenToLogs goroutines which were spawned for the orginal call to AddJob successfully reconnect and so now we have multiple subscriptions which will fire when a log comes in for the same job

### CODE REVIEW

**Review by samsondav — CHANGES_REQUESTED** [review]
Seconded JB's request for a test.

Also, is this thread-safe?

**Comment by connorwstein** [comment]
> Seconded JB's request for a test.
Added 
> Also, is this thread-safe?
It wasn't, although AddJob isn't called from multiple goroutines. I added the locking anyways

**Review by j16r — APPROVED** [review]

**Review by samsondav — APPROVED** [review]
Nice one

File: core/services/job_subscriber.go [hunk 1]
```diff
@@ -89,6 +89,16 @@ func (js *jobSubscriber) Stop() error {
 	return js.jobResumer.Stop()
 }
 
+func (js *jobSubscriber) alreadySubscribed(jobID models.JobID) bool {
+	js.jobsMutex.Lock()
```
Comment (samsondav) [inline core/services/job_subscriber.go:93]: Could be an RLock

**Review by samsondav — APPROVED** [review]

**Pull request merged**